### DISCUSSION TITLE
Does Create 3 include a mapping API?

### DISCUSSION THREAD

**Discussion by dcz-xyz** [iRobotEducation/create3_docs#241] [Questions]
Hi everyone, 

I am working on a robotics project where I can generate a map of a given space which I can then use to designate a location for the robot to move to. Can the Create 3 do this out of the box or do I need to add additional sensing? The robot doesn't necessarily need to respond to dynamic scenarios like people, and I can assume the environment is relatively static. 

Thank you!

**Comment by shamlian** [comment]
Hello!

Create 3 does not have a mapping API; it is a mobility and sensing platform. You'll have to add additional computation, and likely additional sensors, to generate a map. However, if you just need to execute pre-computed moves, you can likely get away with using just the on-board odometry.

Hope this helps.
